src/main.rs

```diff
@@ -1,27 +1,60 @@
 extern crate clap;
+extern crate structopt;
 
+use std::path::PathBuf;
+use structopt::StructOpt;
 use clap::{Arg, App};
 // use std::path::Path;
 // use std::process;
 // use std::fs::File;
 // use std::io::{Read};
 
-mod deamon_lib;
+mod daemon_lib;
+
+/// Struct for commands
+#[derive(StructOpt, Debug)]
+#[structopt(name = "commands")]
+struct Opt {
+    /// Launchd-cli type of command
+    #[structopt(name="METHOD")]
+    method: String,
+
+    /// Script to run
+    #[structopt(name = "FILE", parse(from_os_str))]
+    files: Option<PathBuf>,
+
+    /// Launchd task name
+    #[structopt(short = "n", long = "name", parse(from_os_str))]
+    output: Option<PathBuf>,
+
+    /// Launchd task time
+    #[structopt(short = "every", long = "every")]
+    time: Option<i32>
+}
 
 fn main() {
-    let matches = App::new("launchd-cli")
-      .version("0.1.0")
-      .author("MMW")
-      .about("A launchd cli tool written in Rust")
-      .arg(Arg::with_name("script")
-            .help("script to set.")
-            .empty_values(false)
-        )
-      .get_matches();
+    let opt = Opt::from_args();
+    println!("{:?}", opt); 
     
-    if let Some(script_location) = matches.value_of("script") {
-        if deamon_lib::create() {
-            println!("{}", script_location);
+    match opt.method.as_ref() {
+        "create" => {
+            // TODO: pass values to create method
+            // TODO: create /file(.sh/.py/executable) every 10 min/sec/hours 
+            if daemon_lib::create() {
+                println!("creating");
+            }
+        },
+        "list" => {
+            println!("listing");
```

Comment:
Instead of `println!`, it'd be better to add `unimplemented!()` whenever something isn't yet implemented, this goes for all match branches.

Reply:
good thing to know ;)